kf_lib_data_ingest/etl/load/message_packer.py

```diff
@@ -0,0 +1,714 @@
+        :param target_schema: the target service concept schemas
+        :returns target_instances: Updated version of the input
+        """
+        version = target_schema.get("version")
+        schemas = target_schema.get("definitions")
+        self.logger.info(
+            f"Do null processing using target schema "
+            f'{target_schema.get("target_service")}, '
+            f"version {pformat(version)}"
+        )
+
+        for target_concept, list_of_instances in target_instances.items():
+            # Get schema for target concept
+            schema = schemas.get(target_concept)
+            if not schema:
+                self.logger.info(
+                    f"Skip handle nulls for {target_concept}. No schema "
+                    "was found."
+                )
+                continue
+            # Convert nulls
+            conversions = dict()
+            for i, instance in enumerate(list_of_instances):
+                for attr, value in instance["properties"].items():
+                    if value is not None:
+                        continue
+                    if attr in conversions:
+                        instance["properties"][attr] = conversions[attr]
+                        continue
+
+                    property_def = schema["properties"].get(attr)
+                    mapped_value = None
+                    if not property_def:
+                        self.logger.warning(
+                            "No property definition found for "
+                            f"{target_concept}.{attr} in target schema "
+                            "This property may not exist "
+                            "anymore in the target service."
+                        )
+                    else:
+                        if property_def["type"] == "string":
+                            mapped_value = constants.COMMON.NOT_REPORTED
+                            # Note - the dataservice should tell us if the
+                            # format of the string is `date-time`, then we
+                            # could do:
+                            #   property_def.get('format') == 'date-time'.
+                            # Right now it doesn't do this.
+                            # That's what we should use to determine what null
+                            # value to use. For now, we're just using the name
+                            # of the attribute
+                            if "date" in attr:
+                                mapped_value = None
+                        else:
+                            mapped_value = None
+
+                    self.logger.debug(
+                        f"Mapping {target_concept}.{attr} value {value} to "
+                        f"{mapped_value}"
+                    )
+                    conversions[attr] = mapped_value
+
+                    instance["properties"][attr] = mapped_value
+
+    # TODO: Eliminate these complex Unique Key generation functions.
```

Comment:
Maybe yes, but I need to understand this better. But yea, maybe in another PR we can refactor this